lib/matplotlib/axis.py

```diff
@@ -1052,50 +1052,8 @@ def _update_ticks(self, renderer):
                     ihigh = locs[-1]
             tick_tups = [ti for ti in tick_tups if ilow <= ti[1] <= ihigh]
 
-        # so that we don't lose ticks on the end, expand out the interval ever
```

Comment:
`interval_contains` below is modified to accept floating point slop, so this doesn't need to be here any more.